src/chains.rs

```diff
@@ -351,17 +366,17 @@ impl<Device: DeviceAPI> SwapChainData<Device> {
         let mut color_mask = [0, 0, 0, 0];
         let mut depth_mask = [0];
         let mut stencil_mask = [0];
-        let scissor_enabled = gl.is_enabled(gl::SCISSOR_TEST);
-        let rasterizer_enabled = gl.is_enabled(gl::RASTERIZER_DISCARD);
+        let scissor_enabled = unsafe { gl.is_enabled(gl::SCISSOR_TEST) };
+        let rasterizer_enabled = unsafe { gl.is_enabled(gl::RASTERIZER_DISCARD) };
         unsafe {
-            gl.get_integer_v(gl::DRAW_FRAMEBUFFER_BINDING, &mut bound_fbos[0..]);
-            gl.get_integer_v(gl::READ_FRAMEBUFFER_BINDING, &mut bound_fbos[1..]);
-            gl.get_float_v(gl::COLOR_CLEAR_VALUE, &mut clear_color[..]);
-            gl.get_float_v(gl::DEPTH_CLEAR_VALUE, &mut clear_depth[..]);
-            gl.get_integer_v(gl::STENCIL_CLEAR_VALUE, &mut clear_stencil[..]);
-            gl.get_boolean_v(gl::DEPTH_WRITEMASK, &mut depth_mask[..]);
-            gl.get_integer_v(gl::STENCIL_WRITEMASK, &mut stencil_mask[..]);
-            gl.get_boolean_v(gl::COLOR_WRITEMASK, &mut color_mask[..]);
+            gl.get_parameter_i32_slice(gl::DRAW_FRAMEBUFFER_BINDING, &mut bound_fbos[0..]);
+            gl.get_parameter_i32_slice(gl::READ_FRAMEBUFFER_BINDING, &mut bound_fbos[1..]);
+            gl.get_parameter_f32_slice(gl::COLOR_CLEAR_VALUE, &mut clear_color[..]);
+            gl.get_parameter_f32_slice(gl::DEPTH_CLEAR_VALUE, &mut clear_depth[..]);
+            gl.get_parameter_i32_slice(gl::STENCIL_CLEAR_VALUE, &mut clear_stencil[..]);
+            gl.get_parameter_i32_slice(gl::DEPTH_WRITEMASK, &mut depth_mask[..]);
+            gl.get_parameter_i32_slice(gl::STENCIL_WRITEMASK, &mut stencil_mask[..]);
+            gl.get_parameter_i32_slice(gl::COLOR_WRITEMASK, &mut color_mask[..]);
```

Comment:
this is wrong, GLboolean != int so we need https://github.com/grovesNL/glow/pull/304